tests - Initial checkin for luminous point-to-point upgrade

yuriw opened the pull request:
This suite will test `12.2.1` => `<tip of luminous>` => `-x`

@jdurgin @liewegas I added a dummy `rados/test-upgrade-v12.0.0.sh` (copy from `jewel`) and it needs revising 
@dillaman can you pls take a look at `rbd` related workunuits?

Fixes http://tracker.ceph.com/issues/22048
Signed-off-by: Yuri Weinstein <yweinste@redhat.com>

yuriw commented:
sanity test http://pulpito.ceph.com/teuthology-2017-11-06_20:41:57-upgrade:luminous-x:point-to-point-x-master-distro-basic-ovh/

dillaman commented on qa/suites/upgrade/luminous-x/point-to-point-x/point-to-point-upgrade.yaml:
```diff
@@ -0,0 +1,229 @@
+         - rados/test.sh
+         - cls
+       env:
+         CLS_RBD_GTEST_FILTER: '*:-TestClsRbd.mirror_image'
```
Nit: we should be able to drop this environment override for the luminous point-to-point upgrade. This was a workaround for jewel

yuriw replied:
done

dillaman commented on qa/suites/upgrade/luminous-x/point-to-point-x/point-to-point-upgrade.yaml:
```diff
@@ -0,0 +1,229 @@
+         - rados/test-upgrade-v12.0.0.sh
+         - cls
+       env:
+         CLS_RBD_GTEST_FILTER: '*:-TestClsRbd.mirror_image'
```
Nit: same here

yuriw replied:
removed

jdurgin commented on qa/workunits/rados/test-upgrade-v12.0.0.sh:
```diff
@@ -0,0 +1,53 @@
+
+pids=""
+for f in \
+    'api_aio --gtest_filter=-LibRadosAio.RacingRemovePP:*WriteSame*:*CmpExt*:*RoundTrip3*:*RoundTripPP3*:*Quota*' \
```
I don't think we need a separate test-upgrade-v12.0.0.sh yet

yuriw replied:
@jdurgin which one do we use then ?

jdurgin commented on qa/suites/upgrade/luminous-x/point-to-point-x/point-to-point-upgrade.yaml:
```diff
@@ -0,0 +1,225 @@
+       daemons: [osd.5]
+       wait-for-healthy: false
+       wait-for-up-osds: true
+   # assuming line below can be removed
```
yes it can be removed

jdurgin commented on qa/suites/upgrade/luminous-x/point-to-point-x/point-to-point-upgrade.yaml:
```diff
@@ -0,0 +1,225 @@
+       branch: luminous
+       clients:
+         client.1:
+         ###- rados/test-upgrade-v12.0.0.sh
```
can just use rados/test.sh for all these

yuriw replied:
ack thx !

yuriw commented:
Addressed comments and testing again
http://pulpito.ceph.com/yuriw-2017-11-20_17:30:23-upgrade:luminous-x:point-to-point-x-master-distro-basic-ovh/

yuriw commented:
http://pulpito.ceph.com/teuthology-2017-12-13_16:59:31-upgrade:luminous-x:point-to-point-x-master-distro-basic-ovh/

yuriw commented:
http://tracker.ceph.com/issues/22462
